### PR TITLE
New target flywoo gn f405

### PR DESCRIPTION
**Pull request by mamariomiamo**
@matthewoots and I would like to add support for Flywoo goku-gnf405 series board. 

We have personally tested the firmware on a GOKU GN 405S 20A AIO board with the following features:
1. Serial RX receiver (Frsky-rxsr) on UART1
2. GPS module (MATEKSYS M8Q-5883) on UART6
3. External compass (MATEKSYS M8Q-5883) on I2C
4. BMP280 barometer on I2C
5. Offboard control via UART4
6. Voltage sensing

We flied in VICON with feature 1, 4, 5, 6.

This would enable PX4 running on low-budget micro-sized quadrotor. Possibly beneficial to bring PX4 to more developers.

Lists of available goku gn f405 boards are:
[gn405s 20A AIO](https://flywoo.net/products/goku-gn405s-20a-aio-mpu6000-25-5-x-25-5?_pos=14&_sid=4ea6a68f5&_ss=r)
[gn405s 40A AIO](https://flywoo.net/products/goku-gn405s-40a-aio-mpu6000-25-5-x-25-5?_pos=17&_sid=4ea6a68f5&_ss=r)
[gn405 nano](https://flywoo.net/products/c702-goku-gn405-nano-13a-stack-w-4pcs-ws2812-led-16-16?variant=43201333854448)
[gn405 2S 12A AIO (elrs and no-RX)](https://flywoo.net/products/goku-versatile-f405-2s-12a-aio-w-build-in-elrs-2-4g-rx-mpu6000?_pos=2&_sid=f13924818&_ss=r&variant=43300478189808)

### CODE REVIEW

**Comment by dagar** [comment]
Looks fine, but I haven't reviewed in detail. If you have a flight log please share it and I'll do a quick check for any obvious issues.

Is there a schematic or pinout you can share for this board?

**Comment by mamariomiamo** [comment]
> Looks fine, but I haven't reviewed in detail. If you have a flight log please share it and I'll do a quick check for any obvious issues.
> 
> Is there a schematic or pinout you can share for this board?
The wiring diagram can be found here: https://cdn.shopify.com/s/files/1/0010/7410/2324/files/20210924230948.png?v=1632496369
image
Maybe the betaflight config [here](https://github.com/betaflight/unified-targets/blob/master/configs/default/FLWO-FLYWOOF405S_AIO.config) can provide some insights about the pinout.

Unfortunately these AIO boards do not come with SD logging capabilities, we tried to use mavlink logging via QGC when using mavros but also failed to do so. Maybe we can try with a WiFi telemetry for mavlink logging if it is required.

File: boards/flywoo/gn-f405/init/rc.board_sensors [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+board_adc start
+
+mpu6000 -R 6 -s start
+
+# Start Baro
+bmp280 -X start
+
+# Try to start GPS and compass
+qmc5883l -X start
```
[the remark this answers is not in the document]
Comment (mamariomiamo) [reply boards/flywoo/gn-f405/init/rc.board_sensors]: Updated accordingly.

**Comment by dagar** [comment]
The bootloader bin naming needs to be `${PX4_BOARD_VENDOR}_${PX4_BOARD_MODEL}_bootloader.bin` if you want the `SYS_BL_UPDATE` param and other command line helpers in the build system to automatically work.

**Comment by dagar** [comment]
This is harmless to merge as is (self contained), we can update the bootloader as needed later. Thanks!

**Comment by mamariomiamo** [comment]
> This is harmless to merge as is (self contained), we can update the bootloader as needed later. Thanks!

thanks!

**Pull request merged**